api/ban.go

```diff
@@ -94,11 +94,16 @@ func handleBanImpl(w http.ResponseWriter, r *http.Request) (*database.User, bool
 		return nil, false, "Failed to ban user", http.StatusInternalServerError
 	}
 
-	if req.Tos {
-		gpcm.KickPlayer(req.Pid, "banned")
-	} else {
-		gpcm.KickPlayer(req.Pid, "restricted")
-	}
+	gpcm.KickPlayerCustomMessage(req.Pid, req.Reason, gpcm.WWFCErrorMessage{
+		ErrorCode: 22002,
+		MessageRMC: map[byte]string{
+			gpcm.LangEnglish: "" +
+				"You have been banned from Retro WFC\n" +
+				"Reason: " + req.Reason + "\n" +
+				"Error Code: %[1]d\n" +
+				"Support Info: NG%08[2]x",
```

Comment:
Now that I think of it, we can probably get rid of the support code as no one uses it anywah

Reply:
With the way WWFCErrorMessage is handled, there needs to be somewhere for it to replace [1] and [2] with certain info, and I'd prefer not to make large changes to the codebase for something so simple.
![image](https://github.com/user-attachments/assets/4bfcf611-3a0f-4556-87ae-c04fa112af98)